qcodes/utils/validators.py

```diff
@@ -575,18 +589,21 @@ def validate(self, value: np.ndarray, context: str='') -> None:
     def __repr__(self) -> str:
         minv = self._min_value if math.isfinite(self._min_value) else None
         maxv = self._max_value if math.isfinite(self._max_value) else None
+        # we don't want the repr to execute any deferred shape argument
+        # so we directly use _shape
```

Comment:
```suggestion
        # so we directly use `_shape` through `shape` property
```
right?

Reply:
The comment is out of date and should be removed